### DISCUSSION TITLE
REALITY 的 shortID 和 x255119 密钥是不是有点多余？

### DISCUSSION THREAD

**Discussion by ghost** [XTLS/Xray-core#1887] [Q&A]
本来 VLESS 的 uuid 就有认证、区分用户的功能；而 VMess 的 uuid 还能用来加密。能不能 Xray 内置一套算法，类似于 uuid5 的映射，用 uuid 计算出唯一对应的 shortID 和 x25519 私钥，再通过私钥计算出公钥。这样子不必产生太多参数而造成很多麻烦，也充分发挥 uuid 的作用。

**Comment by RPRX** [comment]
不多余，文章会说明
